include/seqan3/std/ranges

```diff
@@ -230,4 +233,35 @@ using SEQAN3_DOXYGEN_ONLY(iter_move =) ::ranges::iter_move;
 //!\}
 } // namespace std::ranges
 
+// -----------------------------------------------------------------------------------------------------
+// enable_view alias
+// -----------------------------------------------------------------------------------------------------
+
+/* The ranges::enable_view boolean indicates whether a type is a View. In order to create an alias
+ * for the std library that we can specialise for seqan3 types (we cannot specialise using declarations)
+ * we create a new std::ranges::enable_view boolean. Additionally, we now need an ranges::enable_view overload
+ * that delegates to std::ranges::enable_view<type>.
+ */
+
+namespace std::ranges
+{
+
+//!\brief Type trait that indicates whether `type` is a view.
+template <typename type>
+constexpr bool enable_view = ::ranges::detail::enable_view_impl_(::ranges::detail::nullptr_<type>(0));
+
+} // namespace std::ranges
+
+namespace ranges
+{
+
+//!\brief Type trait that indicates whether `type` is a view.
+template <typename type>
+//!\cond
+    requires true
+//!\endcond
+constexpr bool enable_view<type> = std::ranges::enable_view<type>;
+
```

Comment:
I thought about this again, and in this model `std::ranges::enable_view` will not return to for local overloads of `::ranges::enable_view`. 

I think the only solution to this is:
  * declare `std::ranges::enable_view` a plain alias for `::ranges::enable_view` here
  * define `seqan3::enable_view` somewhere in our ranges module, but define it as `int` with default value of `-1`
  * in the same file overload `std::ranges::enable_view` with constraint `seqan3::enable_view == 0 || == 1` and then set it to the corresponding value
  * whenever we want to override a value, we just specify `seqan3::enable_view` to be 0 or 1.

It's not pretty, but it's the only thing that will work I think.

Reply:
:weary: